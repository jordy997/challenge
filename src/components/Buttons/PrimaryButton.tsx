import React from 'react'
import { Text, StyleSheet, View, Dimensions, TouchableWithoutFeedback, GestureResponderEvent } from 'react-native'

type ButtonProps = {
    text: string,
    width: number,
    onPress?: (event: GestureResponderEvent) => void
}
const PrimaryButton: React.FC<ButtonProps> = (props) => {
    const { text, width } = props
    return (
        <View style={[styles.button, { width: width * 2.5 }]}>
            <TouchableWithoutFeedback onPress={() => alert('Tocaste el botÃ²n')}>
                <Text style={styles.text}>{text}</Text>
            </TouchableWithoutFeedback>
        </View>
    )
}

const styles = StyleSheet.create({
    button: {
        width: Dimensions.get('screen').width * 0.9,
        alignSelf: 'center',
        backgroundColor: '#334FFA',
        borderRadius: 5,
        height: 35,
        justifyContent: 'center'
    },
    text: {
        color: 'white',
        textAlign: 'center',
        fontSize: 16,
        fontWeight: '600'
    }
})

export default PrimaryButton