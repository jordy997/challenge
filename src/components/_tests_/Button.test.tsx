import React from 'react'

import { shallow } from 'enzyme'
import PrimaryButton from '../Buttons/PrimaryButton'

const onPress = jest.fn()

let wrapper
describe('Button component', () => {

    beforeEach(() => {
        wrapper = shallow(<PrimaryButton text='Todos' width={130} onPress={() => onPress} />)
    })

    describe('Rendering', () => {
        it('Button should render correctly', () => {
            expect(wrapper).toMatchSnapshot()
        })
    })
})