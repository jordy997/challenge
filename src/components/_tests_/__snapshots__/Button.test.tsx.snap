// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Rendering Button should render correctly 1`] = `ShallowWrapper {}`;
